### PR TITLE
Unite web and mobile welcome page

### PR DESCRIPTION
**Pull request by lyubomir**
Supersedes https://github.com/jitsi/jitsi-meet/pull/1177.

### CODE REVIEW

File: react/features/welcome/components/AbstractWelcomePage.js [hunk 1]
```diff
@@ -34,15 +35,32 @@ export class AbstractWelcomePage extends Component {
...
         };
 
         // Bind event handlers so they are only bound once for every instance.
-        this._onJoinClick = this._onJoinClick.bind(this);
+        const roomnameChanging = this._animateRoomnameChanging.bind(this);
+
+        this._onJoin = this._onJoin.bind(this);
         this._onRoomChange = this._onRoomChange.bind(this);
+        this._updateRoomname = this._updateRoomname.bind(this);
+        this._animateRoomnameChanging = roomnameChanging.bind(this);
```
Comment (lyubomir) [inline react/features/welcome/components/AbstractWelcomePage.js]: @BeatC, what's the point of the following?

````javascript
const roomnameChanging = this._animateRoomnameChanging.bind(this);
this._animateRoomnameChanging = roomnameChanging.bind(this);
````

Isn't that effectively the following?

````javascript
this._animateRoomnameChanging = this._animateRoomnameChanging.bind(this).bind(this);
````

Comment (BeatC) [inline react/features/welcome/components/AbstractWelcomePage.js]: Seems like it's an error. Should be like that:
```
const roomnameChanging = this._animateRoomnameChanging.bind(this);
this._animateRoomnameChanging = roomnameChanging;
```

Comment (lyubomir) [inline react/features/welcome/components/AbstractWelcomePage.js]:
````javascript
this._animateRoomnameChanging
    = this._animateRoomnameChanging.bind(this);
````

File: react/features/welcome/components/WelcomePage.web.js [hunk 2]
```diff
@@ -1,11 +1,112 @@
...
+            enableWelcomePage
+        });
+    }
+
+    /**
+    * Returns the domain name.
+    *
+    * @private
+    * @returns {string} Domain name.
+    **/
```
Comment (lyubomir) [inline react/features/welcome/components/WelcomePage.web.js]: That's no the way we format jsdocs:

````javascript
/**
*
**/
````

The way we format them is:
````javascript
/**
 *
 */
````

Please make sure to comply with the conventions of the existing source code in the future.

File: react/features/welcome/components/WelcomePage.web.js [hunk 3]
```diff
@@ -1,11 +1,112 @@
...
+    _getDomain() {
+        return `${window.location.protocol}//${window.location.host}/`;
+    }
+
+    /**
+    * This method is executed when comonent is mounted.
+    *
+    * @inheritdoc
+    */
```
Comment (lyubomir) [inline react/features/welcome/components/WelcomePage.web.js]: 1. That's not the way we format jsdoc comments.
2. You've introduced 2 different non-compliant jsdoc comment formats in 1 file.

File: react/features/welcome/components/WelcomePage.web.js [hunk 4]
```diff
@@ -36,6 +137,17 @@ export default class WelcomePage extends Component {
...
     /**
+    * Sets input element as property of class.
+    *
+    * @param {HTMLInputElement} input - input element to be set.
+    * @returns {void}
+    * @private
+    **/
+    _setInput(input) {
+        this.roomNameInput = input;
+    }
```
Comment (lyubomir) [inline react/features/welcome/components/WelcomePage.web.js]: I'm afraid I find the business with _setInput and _roomNameInput unnecessary and I'll remove them.

File: react/features/welcome/components/WelcomePage.web.js [hunk 5]
```diff
@@ -139,6 +255,115 @@ export default class WelcomePage extends Component {
...
+            );
+        }
+
+        return null;
+    }
+
+    /**
+    * Renders powered by block if it is enabled
+    *
+    * @returns {void}
```
Comment (lyubomir) [inline react/features/welcome/components/WelcomePage.web.js]: The `@returns` type of `_renderPoweredBy` is incorrect.

File: react/features/welcome/components/WelcomePage.web.js [hunk 6]
```diff
@@ -1,11 +1,112 @@
...
+
+import React from 'react';
+import { connect } from 'react-redux';
+
+import {
+    AbstractWelcomePage,
+    mapStateToProps
+} from './AbstractWelcomePage';
+
+const RIGHT_WATERMARK_STYLES = {
```
Comment (lyubomir) [inline react/features/welcome/components/WelcomePage.web.js]: Our coding style states that we require jsdocs on everything.

**Pull request merged**